### PR TITLE
luaformatter: fix darwin build

### LINKED ISSUE #136290: luaformatter doesn't build on MacOS
### Describe the bug
`luaformatter` doesn't build on MacOS

### Steps To Reproduce
1. `nix-shell -p luaformatter` on MacOS

### Expected behavior
Enter nix shell with `luaformatter`

### Screenshots
N/A

### Additional context
Build log: https://pastebin.com/qRPQ3maZ
This is probably blocked by: https://github.com/NixOS/nixpkgs/issues/101229

### Notify maintainers

@figsoda @SuperSandro2000

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.6.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.14`
 - channels(jupblb): `"home-manager, nixpkgs-21.11pre312335.8d8a28b47b7"`
 - nixpkgs: `/Users/jupblb/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by figsoda**
###### Motivation for this change

fix #136290

I am not able test on darwin, but the same change fixes a similar issue [here](https://github.com/NixOS/nixpkgs/pull/119018#issuecomment-823045731).
I would be great if someone can test it out for me 

@jupblb @SuperSandro2000 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jupblb** [comment]
Switching to `gccStdenv` doesn't work after https://github.com/NixOS/nixpkgs/pull/134716. I guess one solution would be to revert it and merge this PR in order to support both platforms.

**Comment by figsoda** [comment]
Not sure what is causing this issue.
on #123530 it [builds on darwin](https://github.com/NixOS/nixpkgs/pull/123530/checks?check_run_id=2621381006)
on #134669 it [fails on darwin](https://github.com/NixOS/nixpkgs/pull/134669/checks?check_run_id=3363465863)
There is no changes to luaformatter between these commits, might be changes to `stdenv` that is causing the issue

**Comment by figsoda** [comment]
@ofborg build luaformatter

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -14127,10 +14127,7 @@ with pkgs;
 
   lttv = callPackage ../development/tools/misc/lttv { };
 
-  luaformatter = callPackage ../development/tools/luaformatter
-    (lib.optionalAttrs stdenv.isDarwin {
-      stdenv = overrideCC stdenv llvmPackages_latest.clang;
-    });
+  luaformatter = callPackage ../development/tools/luaformatter { };
```
Comment (marsam) [inline pkgs/top-level/all-packages.nix:14130]: I think you can fix the build with:
```suggestion
  luaformatter = callPackage ../development/tools/luaformatter {
    stdenv = if stdenv.cc.isClang then llvmPackages_9.stdenv else stdenv;
  };
```

Comment (figsoda) [reply pkgs/top-level/all-packages.nix:14130]: it works! thanks

weird that llvmPackages_latest doesn't work

**Comment by happysalada** [comment]
Just saying I'm running nixpkgs-review on darwin right now.

**Comment by happysalada** [comment]
Thank you for the contribution!

**Pull request merged**